src/core/models.py

```diff
@@ -1919,3 +1968,641 @@ def log_hijack_ended(sender, hijacker, hijacked, request, **kwargs):
+        institution='',
+        department='',
+        country='',
+        account=None,
+        frozen_author=None,
+        preprint_author=None,
+    ):
+        """
+        Backwards-compatible API for setting affiliation from unstructured text.
+        Intended for use in batch importers where ROR data is not available.
+        Does not support ROR ids, multiple affiliations, or start or end dates.
+        When possible, include department and country to create unified records.
+        Only include one of author, frozen_author, or preprint_author.
+
+        :param institution: the uncontrolled organization name as a string
+        :type institution: str
+        :type department: str
+        :param country: ISO-3166-1 alpha-2 country code or Janeway Country object
+        :type country: str or core.models.Country
+        :type account: core.models.Account
+        :type frozen_author: submission.models.FrozenAuthor
+        :type preprint_author: repository.models.PreprintAuthor
+        """
+        organization, _created = Organization.get_or_create_without_ror(
+            institution=institution,
+            country=country,
+            account=account,
+            frozen_author=frozen_author,
+            preprint_author=preprint_author,
+        )
+
+        # Create or update the actual affiliation if the associated
+        # account / frozen author / preprint author has been saved already
+        try:
+            affiliation, created = Affiliation.objects.get_or_create(
+                account=account,
+                frozen_author=frozen_author,
+                preprint_author=preprint_author,
+                organization=organization,
+                department=department,
+            )
+            affiliation.is_primary = True
+            affiliation.save()
+        except ValueError:
+            logger.warn('The affiliation could not be created.')
+            affiliation = None
+            created = False
+        return affiliation, created
+
+    @classmethod
+    def naive_set_primary_department(
+        cls,
+        value,
+        account=None,
+        frozen_author=None,
+    ):
+        """
+        Backwards-compatible API for setting department names in isolation.
+        It is better to use Affiliation.get_or_create_without_ror where department
+        is set togther with institution and country.
+        Does not support ORCIDs, RORs or multiple affiliations.
+        """
+        # Create or update an affiliation if the associated
+        # account / frozen author / preprint author has been saved already
+        try:
+            affiliation, _created = Affiliation.objects.get_or_create(
+                account=account,
+                frozen_author=frozen_author,
+                is_primary=True,
+            )
+            affiliation.department = value
+            affiliation.save()
+        except ValueError:
+            logger.warn('The department could not be set.')
```

Comment:
if we use a warning level, we need more context on this message (e.g.   `logger.warn('The department could not be set to "%s"' % value)`